class-wp-sw-manager.php

```diff
@@ -126,14 +130,24 @@ private function default_scope() {
         }
 
         private function setup_sw_registrar_script() {
-            $this->router->add_route(self::SW_REGISTRAR_SCRIPT_URL,  array($this, 'sw_registrar'));
+            $this->dynamic_server->add_file(self::SW_REGISTRAR_SCRIPT_URL,  array($this, 'sw_registrar'));
+            add_action('init', array($this, 'check_registrations'), 999);
             add_action('wp_enqueue_scripts', array($this, 'enqueue_registrar'));
         }
 
+        public function check_registrations() {
+            $last_registrations = get_option('wpswmanager_registrations', array());
+            $current_registrations = array_keys($this->service_workers);
+            $areEqual = array_count_values($last_registrations) == array_count_values($current_registrations);
+            if (!$areEqual) {
```

Comment:
Also, the files should be invalidated whenever the content of the service worker changes, not only when a new registration is added.